utils.py

```diff
@@ -72,6 +74,65 @@ def mutations_data(self):
             }
         )
 
+    @staticmethod
+    @numba.njit
+    def calc_mean_tree_arity(num_children_per_tree, num_nodes_per_tree, num_leaves):
+        return num_children_per_tree / (num_nodes_per_tree - num_leaves)
+
+    @staticmethod
+    @numba.njit
+    def calc_sum(a):
+        return np.sum(a)
+
+    @staticmethod
+    @numba.njit
+    def calc_max(a):
+        return np.max(a)
+
+    @staticmethod
+    @numba.njit
+    def get_size(a):
+        return a.size
+
+    def trees_data(self):
+        ts = self.ts
+        num_children_per_tree = np.zeros(ts.num_trees)
+        num_nodes_per_tree = np.zeros(ts.num_trees)
+        num_leaves_per_tree = ts.num_samples
+        max_internal_arity = np.zeros(ts.num_trees)
+        total_branch_length = np.zeros(ts.num_trees)
+
+        for tree in ts.trees():
+            num_children_per_node = tree.num_children_array[:-1]
```

Comment:
There's not much point in these functions like calc_sum etc, as just replace with np.sum